### PR TITLE
Set search bar width to placeholder text

### LINKED ISSUE #665: Allow setting the width of search bar
The problem:

image

### PR DESCRIPTION
**Pull request by weiquu**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [x] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

Resolves #665 

**Overview of changes:**
Binds the width of the search bar to the length of the placeholder text so as to avoid clipping of the placeholder text.

Before:
search bar before

After:
search bar after

**Anything you'd like to highlight/discuss:**
The width is set using the `ch` unit, which is defined according to the width of the `o` character in a given typeface. Hence, it might run into issues with certain non-standard typefaces, should the MarkBind typeface be changed in the future.

**Testing instructions:**
Use a long placeholder text to check if the width of the search bar expands accordingly.

**Proposed commit message: (wrap lines at 72 characters)**
Set search bar width to placeholder text

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [ ] Linked all related issues
- [ ] No unrelated changes

### CODE REVIEW

**Comment by tlylt** [comment]
Hi @weiquu , thank you for contributing to MarkBind! Quick question: if the placeholder is of one character long, will the width be just of 1 character wide?

**Comment by weiquu** [comment]
> Hi @weiquu , thank you for contributing to MarkBind! Quick question: if the placeholder is of one character long, will the width be just of 1 character wide?

Hello! The width will be set to `1ch`, but due to the minimum width of `.form-control` being set to `8em`, it will look something like this (same size as a blank placeholder):
search bar 1 char

File: packages/vue-components/src/Searchbar.vue [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <input
       v-else
       v-model="value"
+      v-bind:style="{width: placeholder.length + 'ch'}"
```
Comment (tlylt) [inline packages/vue-components/src/Searchbar.vue:7]: Try running `npm run lint` and you should see this warning:
>  7:7  warning  Unexpected 'v-bind' before ':'  vue/v-bind-style

If you have ESlint installed you should be able to see a warning and how to resolve it, if you are using VSCode:
image

File: packages/core-web/dist/js/markbind.min.js.LICENSE.txt [hunk 2]
```diff
@@ -8,28 +8,3 @@
   * https://github.com/linusborg/portal-vue
   * 
  */
-
```
Comment (tlylt) [inline packages/core-web/dist/js/markbind.min.js.LICENSE.txt:11]: Please refer to https://markbind-master.netlify.app/devguide/workflow#editing-frontend-features
for the details on how frontend features are updated. I think this file change and other ones in `dist` should not be included in this PR.

**Review by tlylt — COMMENTED** [review]
The width does expand to accommodate the length of the placeholder characters, however the current min-width feels a little too tight if the placeholder is short. For example:
Original:
image
With this change:
image
